File: infrapulse_website/frontend/src/types/queue.ts
import { DefectType, CategoryType, SeverityType, PriorityLevelType, ComplaintStatusType } from './complaint';

export interface QueueItemResponse {
  rank: number;
  id: string;
  complaint_id: string;
  user_id: string;
  name_snapshot: string;
  address: string;
  description: string;
  image_url: string;
  detected_defect?: DefectType | string | null;
  category: CategoryType | string;
  confidence?: number | null;
  visible_extent_percentage?: number | null;
  extent_label?: string | null;
  severity_score?: number | null;
  severity?: SeverityType | string | null;
  priority_score?: number | null;
  priority_level?: PriorityLevelType | string | null;
  status: ComplaintStatusType | string;
  created_at: string;
}

export interface QueueListResponse {
  category: CategoryType | string;
  total_items: number;
  items: QueueItemResponse[];
}

export interface QueuePositionResponse {
  complaint_id: string;
  category?: string | null;
  in_queue: boolean;
  rank?: number | null;
  queue_size?: number | null;
  status: string;
  ai_status: string;
}

File: infrapulse_website/backend/app/schemas/queue.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict

class QueueItemResponse(BaseModel):
    rank: int
    id: str
    complaint_id: str
    user_id: str
    name_snapshot: str
    address: str
    description: str
    image_url: str
    detected_defect: Optional[str] = None
    category: str
    confidence: Optional[float] = None
    visible_extent_percentage: Optional[float] = None
    extent_label: Optional[str] = None
    severity_score: Optional[float] = None
    severity: Optional[str] = None
    priority_score: Optional[float] = None
    priority_level: Optional[str] = None
    status: str
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

class QueueListResponse(BaseModel):
    category: str
    total_items: int
    items: List[QueueItemResponse]

class QueuePositionResponse(BaseModel):
    complaint_id: str
    category: Optional[str] = None
    in_queue: bool
    rank: Optional[int] = None
    queue_size: Optional[int] = None
    status: str
    ai_status: str
